tools/diagnostics/full_bundle.py

```diff
@@ -0,0 +1,169 @@
+from datetime import date, datetime
+from typing import List
+import logging
+import os
+import retrying
+
+import sdk_cmd
+import sdk_utils
+
+from diagnostics.service_bundle import ServiceBundle
+import diagnostics.base_tech_bundle as base_tech
+from diagnostics.bundle import Bundle
+
+log = logging.getLogger(__name__)
+
+DEFAULT_RETRY_WAIT = 1000
+DEFAULT_RETRY_MAX_ATTEMPTS = 5
+
+
+@retrying.retry(wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS)
+def get_dcos_services() -> dict:
+    return sdk_cmd.get_json_output("service --completed --inactive --json", print_output=False)
+
+
+def to_dcos_service_name(service_name: str) -> str:
+    """DC/OS service names don't contain the first slash.
+    i.e.: |     SDK service name     |   DC/OS service name    |
+          |--------------------------+-------------------------|
+          | /data-services/cassandra | data-services/cassandra |
+    """
+    return service_name.lstrip("/")
+
+
+def is_service_named(service_name: str, service: dict) -> bool:
+    return service.get("name") == to_dcos_service_name(service_name)
+
+
+def is_service_active(service: dict) -> bool:
+    return service.get("active") is True
+
+
+def services_with_name(service_name: str, services: List[dict]) -> List[dict]:
+    return [s for s in services if is_service_named(service_name, s)]
+
+
+def active_services_with_name(service_name: str, services: List[dict]) -> List[dict]:
+    return [s for s in services if is_service_named(service_name, s) and is_service_active(s)]
+
+
+def is_service_scheduler_task(package_name: str, service_name: str, task: dict) -> bool:
+    labels = task.get("labels", [])
+    dcos_package_name = next(
+        iter([l.get("value") for l in labels if l.get("key") == "DCOS_PACKAGE_NAME"]), ""
+    )
+    dcos_service_name = next(
+        iter([l.get("value") for l in labels if l.get("key") == "DCOS_SERVICE_NAME"]), ""
+    )
+    return dcos_package_name == package_name and dcos_service_name == to_dcos_service_name(
+        service_name
+    )
+
+
+def directory_date_string() -> str:
+    return date.strftime(datetime.now(), "%Y%m%d%H%M%S")
```

Comment:
Please make this UTC and use an ISO 8601 format.

Reply:
b6947d737c960703a85fa48b98552a6f7cb5ac84